### ISSUE TITLE
Change the size of the preview-element doesn't work

### ISSUE THREAD

**Issue by JanKe0909** [zxing-js/ngx-scanner#55]
Hello,

I tried to change the css-style as explained in usage but it doesn't worked for me.

My html file looks like this below:
`<zxing-scanner #scanner class="test-class" 
      [scannerEnabled]="scannerEnabled"
      [device]="selectedDevice"
      (camerasFound)="displayCameras($event)"
      (scanSuccess)="handleQrCodeResult($event)"
      [cssClass]="'small-video'"></zxing-scanner>`

and I try to change the style like this:

`.small-video {
  max-height: 70vh;
  width: 100vw;
  object-fit: contain;
}`

Thank you and best regards, Jan

**Comment by odahcam** [comment]
Please change the size of the `zxing-scanner` element, the elements inside the component should not be entirely customizable to avoid some extra issues, so we bounded the preview-video size to the `zxing-scanner` element size.

We will update the docs ASAP to avoid this to happen again.

**Comment by andrewyan** [comment]
Hi, I’m a newbie to all these. I guess I’ve the same problem where the preview videos are always larger than the phone screen. Can someone put a simple example here how it can be shrinked to fit the screen? Many thanks!

**Comment by odahcam** [comment]
You will loose the video proportions and the quality if you shrink the video. You **can** change the preview element size via CSS, there's only some existing styles you'll gotta overide. It just shouldn't be that way, but you can do that with some extra effort. Just do what @JanKe0909 did and add the required CSS overides so your styles are aplied to the element.

**Comment by andrewyan** [comment]
I'm really sorry. When I said I'm a newbie, I'm serious. Could you give me an example of these overrides?

**Comment by werthdavid** [comment]
Have a look at [this demo](https://stackblitz.com/edit/zxing-ngx-scanner-layout) 
As you can see I made an scss-file that contains a css-class that resizes the video to the desired size. Be aware that `/deep/` is used in the example due to angulars [view encapsulation](https://angular.io/guide/component-styles) 
an alternative would be to set ViewEncapsulation.None inside the app-component.
Or to make it short: when you use `/deep/` the style of that css-class is valid not only inside your component but through the whole angular-app

If you have any questions don't hesitate to ask

**Comment by odahcam** [comment]
As long as `/deep/` is deprecated, I would like to suggest a change:

```css
zxing-scanner::ng-deep video {
    /* ... */
}
```

In Angular you can use the `::ng-deep` filter instead. 😁 

Working example:
https://stackblitz.com/edit/zxing-ngx-scanner-layout-shvqdr?file=app%2Fapp.component.scss


Also, now that we have this built in feature, I would like to ask if it isn't time to remove the `cssClass` attribute. It could increase a little performance and maintainability, also docs about this could help new guys to find easier answers. What you think David? Maybe a new issue for discussion?

**Comment by andrewyan** [comment]
Hi David & Luis, thanks so much for your detail info. I'll check them out and probably need to learn a lot more about Angular :) You people are very helpful and I really appreciate that!

**Comment by JanKe0909** [comment]
Hey guys, thank you for your help and your fast response. It works for me. 👍 Best regards, Jan

**Comment by csutorasr** [comment]
This could be achieved through an `ngTemplate` as `::ng-deep` is [not a solution](https://github.com/zxing-js/ngx-scanner/issues/60#issuecomment-379088573)

**Comment by csutorasr** [comment]
[Here](https://stackblitz.com/edit/angular-q5mgpg) is the solution.

The video element and the template should be a ref. This way you can still control the video element, but with the template a css from the component is applied to the video element without `ngDeep`.

**Comment by odahcam** [comment]
Yeah, this is a clean solution, but can be difficult for newbies (to use or implement).

This is not something meant to be changed, so I'm starting to think about sizing `@Input`s such as width, height, proportions, etc.

The focus is to customize the `zxing-scanner` and maybe some new properties could solve the video size.

**Comment by werthdavid** [comment]
@csutorasr thanks for the example, looks very clean in deed.
I'd suggest to go with this solution, I think we're making our life harder by introducing more Inputs.
If somebody doesn't understand how to properly use that syntax he can still (for now) use the hacky `/deep/` workaround but I don't like the idea of more complexity with Inputs just b/c some people might not understand it

**Comment by odahcam** [comment]
In that case, I prefer to create some `zxing-scanner`-directive you can put in a `<video>` tag than creating this kinda stuff, the directive is gonna easier to handle by both sides.
